html_sanitizer/sanitizer.py

```diff
@@ -49,11 +49,10 @@ def sanitize_href(href):
 def normalize_overall_whitespace(
     html, keep_typographic_whitespace=False, whitespace_re=None
 ):
-    # remove all sorts of newline and nbsp characters
-    whitespace = ["\n", "&#10;", "&#xa;", "\r", "&#13;", "&#xd;"]
-    if not keep_typographic_whitespace:
-        # non-breaking space representations
-        whitespace += ["\xa0", "&nbsp;", "&#160;", "&#xa0;"]
+    if keep_typographic_whitespace:
+        return html
+        # remove all sorts of newline and nbsp characters
+    whitespace = ["\xa0", "&nbsp;", "&#160;", "&#xa0;","\n", "&#10;", "&#xa;", "\r", "&#13;", "&#xd;"]
     if whitespace_re is None:
         whitespace_re = r"\s+"
```

Comment:
Could you move these two lines down just above the `re.sub` call while you're active in this area?